theano/gof/vm.py

```diff
@@ -890,6 +897,89 @@ def make_all(self, profiler=None, input_storage=None,
             compute_map[k] = [k.owner is None]
 
         thunks = []
+
+        # Collect Reallocation Info
+        compute_map_re = defaultdict(lambda: [0])
+        for var in fgraph.inputs:
+            compute_map_re[var][0] = 1
+
+        viewed_by = {}
+        for var in fgraph.variables:
+            viewed_by[var] = []
+        view_of = {}
+        reallocated_info = {}
+        dependencies = getattr(fgraph.profile, 'dependencies', {})
```

Comment:
Here, we can't use {} as a default. Otherwise, we will produce bad result in the following code. Can you find a way to always have it? If you can't have it, you should disable the reallocation. Also, if you can put as a comment the case where we can't get the dependencies, it would be great.